### PR TITLE
Bug fixes for PE-D

### LINKED ISSUE #344: [PE-D][Tester A] Address can have more than 100 characters
Documentation states the max length of address is 100, however the following command succeeds with the address being 101 characters long.
```
add n|John Doe p|98765432 e|johnd@example.com a|AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
```

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: Dethada/ped#4

### LINKED ISSUE #346: [PE-D][Tester A] Edit command allows address length larger than 100
The documentation states that address has a max length of 100.
The following command sets the target person's address to a string with length larger than 101.
```
edit 1 a|AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
```

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: Dethada/ped#5

### LINKED ISSUE #355: [PE-D][Tester A] NullPointerException thrown when user clicks on the exam
```
Exception in thread "JavaFX Application Thread" java.lang.NullPointerException
        at javafx.controls/com.sun.javafx.scene.control.behavior.ListViewBehavior.mousePressed(ListViewBehavior.java:369)
        at javafx.controls/com.sun.javafx.scene.control.inputmap.InputMap.handle(InputMap.java:274)
        at javafx.base/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:247)
        at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
        at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
        at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
        at javafx.base/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
        at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
        at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
        at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
        at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
        at javafx.graphics/javafx.scene.Scene$MouseHandler.process(Scene.java:3894)
        at javafx.graphics/javafx.scene.Scene.processMouseEvent(Scene.java:1887)
        at javafx.graphics/javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2620)
        at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:411)
        at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:301)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(GlassViewEventHandler.java:450)
        at javafx.graphics/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:424)
        at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:449)
        at javafx.graphics/com.sun.glass.ui.View.handleMouseEvent(View.java:551)
        at javafx.graphics/com.sun.glass.ui.View.notifyMouse(View.java:937)
        at javafx.graphics/com.sun.glass.ui.mac.MacView.notifyMouse(MacView.java:127)
```

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: Dethada/ped#8

### LINKED ISSUE #358: [PE-D][Tester E] Error message does not match error
## Description of the Bug
Known constraint of the length of the value for `s|` is `7 + 2 decimals`. However, error message does not report on the maximum length of the `score` for exam.

## Steps to Reproduce
1. Add an exam with the name as `midterm` and score as `10000000` using `addExam n|midterm s|10000000`

## Expected behavior
Adds an exam with the name as `midterm` and score as `10000000`

## Actual behavior

Error that does not tell me about the limit in maximum score.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: tsulim/ped#2

### LINKED ISSUE #363: [PE-D][Tester E] No error thrown when email does not fulfil constraint
## Description of the Bug
Application accepts the email input when it did not fulfil the constraints for it. The constraint that it did not meet is: `Made up of domain labels followed by periods`

## Steps to Reproduce
1. `add n|test p|98765432 e|1@12 a|normal address`

## Expected behavior
Throws an error message on invalid email

## Actual behavior
Accepts the email and add person

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: tsulim/ped#4

### LINKED ISSUE #364: [PE-D][Tester E] No error thrown when address goes over maximum length
## Description of the Bug
Application accepts the address input when it is more than the `maximum length of 100`

## Steps to Reproduce
1. `add n|test p|98765432 e|test@example.com a|12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901` - 101 characters for address

## Expected behavior
Throws an error message on maximum length of address

## Actual behavior
Accepts the address and add person

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: tsulim/ped#5

### LINKED ISSUE #365: [PE-D][Tester E] Incorrect error message for filtering person
## Description of the Bug
Typos found in displayed error message for `find`

## Steps to Reproduce
1. `find`

## Expected behavior
```
Invalid command format! 
find: Finds all persons with a specified aspect containing specified keyword (case-insensitive) or
which meets the criteria with the specified value (positive-integer)
and displays them as a list with index numbers.
The currently supported prefixes are: n|, p|, e|, a|, t|, m|, r|, s|, lt|, mt|
Parameters: PREFIX|KEYWORD
Example: find n|Alice
```

## Actual behavior
```
Invalid command format! 
find: Finds all persons with a specified aspect containing specified keyword (case-insensitive) or
which meets the criteria with the specified value (positive-integer)
and displays them as a list with index numbers.
The currently supported prefixes are: [n|, p|, e|, a|, t|, m|, r|, s|, lt|, mt|]
Parameters: PREFIX/KEYWORD
Example: find n/Alice
```

-------------
Labels: `type.FunctionalityBug` `severity.VeryLow`
original: tsulim/ped#6

### LINKED ISSUE #366: [PE-D][Tester E] Incorrect error message for adding exam
## Description of the Bug
Typos found in displayed error message for `addExam` and the parameters are not consistent with the format given in the [user guide](https://ay2324s2-cs2103t-t10-1.github.io/tp/UserGuide.html#adding-an-exam-addexam)

## Steps to Reproduce
1. `addExam`

## Expected behavior
```
Invalid command format! 
addExam: Adds an exam to the address book. Parameters: n|NAME s|MAX_SCORE
Example: addExam n|Midterm s|100
```

## Actual behavior
```
Invalid command format! 
addExam: Adds an exam to the address book. Parameters: EXAM_NAME n/EXAM_DETAILS s/MAXIMUM_SCORE
Example: addExam n|Midterm s|100
```

-------------
Labels: `type.FunctionalityBug` `severity.VeryLow`
original: tsulim/ped#7

### LINKED ISSUE #367: [PE-D][Tester E] Typo in error message for adding score
## Description of the Bug
Typos found in displayed error message for `addScore`.

## Steps to Reproduce
1. `addScore`

## Expected behavior
```
Invalid command format! 
addScore: Adds a score to the person identified by the index number used in the last person listing. Parameters: INDEX (must be a positive integer) s|SCORE
Example: addScore 1 s|85
```

## Actual behavior
```
Invalid command format! 
addScore: Adds a score to the person identified by the index number used in the last person listing. Parameters: INDEX (must be a positive integer) s/SCORE
Example: addScore 1 s/85
```

-------------
Labels: `type.FunctionalityBug` `severity.VeryLow`
original: tsulim/ped#8

### PR DESCRIPTION
**Pull request by delishad21**
Couple of bug fixes relating to input validation and UI

Fixes #367, fixes #366, fixes #365, fixes #364, fixes #346, fixes #344, fixes #363, fixes #358, fixes #355

### CODE REVIEW

**Review by Zer0Legion — COMMENTED** [review]
Good changes. They will definitely benefit us in the future as well with the refactoring.

File: src/main/java/seedu/address/logic/commands/AddExamCommand.java
Comment (jayllo-c) [inline src/main/java/seedu/address/logic/commands/AddExamCommand.java:1]: Nice job changing the prefixes to the ones indicated in the CliSyntax file

File: src/main/java/seedu/address/logic/commands/AddScoreCommand.java [hunk 1]
```diff
@@ -27,9 +28,9 @@ public class AddScoreCommand extends Command {
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": Adds a score to the person identified by the index number used in the last person listing. "
             + "Parameters: INDEX (must be a positive integer) "
-            + "s/SCORE\n"
+            + PREFIX_SCORE + "SCORE\n"
```
Comment (jayllo-c) [inline src/main/java/seedu/address/logic/commands/AddScoreCommand.java:31]: Thorough changes

File: src/main/java/seedu/address/ui/NoSelectionModel.java
Comment (jayllo-c) [inline src/main/java/seedu/address/ui/NoSelectionModel.java:1]: solves the issue of null ptr exceptions being thrown

**Review by jayllo-c — COMMENTED** [review]
Nice job ensuring that some issues raised in the PE-D were addressed

**Comment by jayllo-c** [comment]
Codecov issues thrown because UI additions are not covered

**Pull request merged**